src/containerapp/azext_containerapp/_utils.py

```diff
@@ -74,11 +74,14 @@ def process_service(cmd, resource_list, service_name, arg_dict, subscription_id,
 
                 if service_type is None or service_type not in DEV_SERVICE_LIST:
                     raise ResourceNotFoundError(f"The service '{service_name}' does not exist")
-
-                service_bindings_def_list.append({
+                service_bind = {
                     "serviceId": containerapp_def["id"],
-                    "name": binding_name
-                })
+                    "name": binding_name,
+                    "clientType": arg_dict.get("clientType")
+                }
```

Comment:
Will it is called by update command? If yes, when --client-type not provided, will it is cleared

Reply:
No, for update command, we get the value from existing containerapp first, if the input `clientType` not exists, we will not set the value requestBody. 
We have test for `test update customizedKeys without clientType`
![image](https://github.com/Azure/azure-cli-extensions/assets/46143499/8e683480-579c-4090-83da-f484697ca41d)